Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -0,0 +1,431 @@
+
+        unless args.quiet?
+          puts "#{Tty.red}#{formula_name(formula, args: args)}#{Tty.reset} : deprecated"
+          return
+        end
+      end
+
+      if formula.versioned_formula? && !formula.livecheckable?
+        return status_hash(formula, "versioned", args: args) if args.json?
+
+        unless args.quiet?
+          puts "#{Tty.red}#{formula_name(formula, args: args)}#{Tty.reset} : versioned"
+          return
+        end
+      end
+
+      if formula.head? && !formula.any_version_installed?
+        head_only_msg = "HEAD only formula must be installed to be livecheckable"
+        return status_hash(formula, "error", [head_only_msg], args: args) if args.json?
+
+        unless args.quiet?
+          puts "#{Tty.red}#{formula_name(formula, args: args)}#{Tty.reset} : #{head_only_msg}"
+          return
+        end
+      end
+
+      is_gist = formula.stable&.url&.include?("gist.github.com")
+      if formula.livecheck.skip? || is_gist
+        skip_msg = if formula.livecheck.skip_msg.is_a?(String) &&
+                      formula.livecheck.skip_msg.present?
+          formula.livecheck.skip_msg.to_s
+        elsif is_gist
+          "Stable URL is a GitHub Gist"
+        else
+          ""
+        end
+
+        return status_hash(formula, "skipped", (skip_msg.blank? ? nil : [skip_msg]), args: args) if args.json?
+
+        unless args.quiet?
+          puts "#{Tty.red}#{formula_name(formula, args: args)}#{Tty.reset} : skipped" \
+              "#{" - #{skip_msg}" if skip_msg.present?}"
+          return
+        end
+      end
+
+      false
+    end
+
+    # Formats and prints the livecheck result for a formula.
+    # @return [nil]
+    def print_latest_version(info, args:)
+      formula_s = "#{Tty.blue}#{info[:formula]}#{Tty.reset}"
+      formula_s += " (guessed)" if !info[:meta][:livecheckable] && args.verbose?
+
+      current_s = if info[:version][:newer_than_upstream]
+        "#{Tty.red}#{info[:version][:current]}#{Tty.reset}"
+      else
+        info[:version][:current]
+      end
+
+      latest_s = if info[:version][:outdated]
+        "#{Tty.green}#{info[:version][:latest]}#{Tty.reset}"
+      else
+        info[:version][:latest]
+      end
+
+      puts "#{formula_s} : #{current_s} ==> #{latest_s}"
+    end
+
+    # Returns an Array containing the formula URLs that can be used by livecheck.
+    # @return [Array]
+    def checkable_urls(formula)
+      urls = []
+      urls << formula.head.url if formula.head
+      if formula.stable
+        urls << formula.stable.url
+        urls.concat(formula.stable.mirrors)
+      end
+      urls << formula.homepage if formula.homepage
+
+      urls.compact
+    end
+
+    # Preprocesses and returns the URL used by livecheck.
+    # @return [String]
+    def preprocess_url(url)
+      # Check for GitHub repos on github.com, not AWS
+      url.sub!("github.s3.amazonaws.com", "github.com") if url.include?("github")
```

Comment:
```suggestion
      url = url.sub("github.s3.amazonaws.com", "github.com") if url.include?("github")
```

This addresses the `can't modify frozen String` error that we discussed. The error was a good thing in this instance, as it let us know that we're doing something that we don't actually want to be doing (i.e., doing an in place modification).